nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -90,20 +90,55 @@ public String toString() {
         }
     }
 
+    // validator to ensure that valid EL expressions are used in the directory property
+    static final Validator PATH_WITH_EL_VALIDATOR = new Validator() {
```

Comment:
Definitely agree this validator is not really specific to the HDFS processors and could be moved to the StandardValidators. Let me work on doing that. 

The one that is already there expects only a single attribute expressions like "${hadoop.dir}"  and fails validation for something like "data/${hadoop.dir}" so that was why I had to create another one that uses newExpressionLanguageCompiler().validateExpression(input, true)  instead of newExpressionLanguageCompiler().compile(input)